exec/java-exec/src/test/java/org/apache/drill/exec/record/vector/TestValueVector.java

```diff
@@ -223,14 +231,187 @@ public void testNullableVarLen2() {
+      }
+    }
+*/
+
+    vector1.close();
+/* TODO(cwestin)
+    vector2.close();
+    buffer1.release();
+*/
+  }
+
+  @Test
+  public void testVarCharVectorLoad() {
+    final MaterializedField field = MaterializedField.create(EMPTY_SCHEMA_PATH, VarCharHolder.TYPE);
+
+    // Create a new value vector for 1024 variable length strings.
+    final VarCharVector vector1 = new VarCharVector(field, allocator);
+    final VarCharVector.Mutator mutator = vector1.getMutator();
+    vector1.allocateNew(1024 * 10, 1024);
+
+    // Populate the vector.
+    final StringBuilder stringBuilder = new StringBuilder();
+    final int valueCount = 10;
+    for(int i = 0; i < valueCount; ++i) {
+      stringBuilder.append('x');
+      mutator.setSafe(i, stringBuilder.toString().getBytes(utf8Charset));
+    }
+    mutator.setValueCount(valueCount);
+    assertEquals(valueCount, vector1.getAccessor().getValueCount());
+
+    // Combine the backing buffers so we can load them into a new vector.
+    final DrillBuf[] buffers1 = vector1.getBuffers(false);
+    final DrillBuf buffer1 = combineBuffers(allocator, buffers1);
+    final VarCharVector vector2 = new VarCharVector(field, allocator);
+    vector2.load(vector1.getMetadata(), buffer1);
+
+    // Check the contents of the new vector.
+    final VarCharVector.Accessor accessor = vector2.getAccessor();
+    stringBuilder.setLength(0);
+    for(int i = 0; i < valueCount; ++i) {
+      stringBuilder.append('x');
+      final Object object = accessor.getObject(i);
+      System.out.println(object);
+      assertEquals(stringBuilder.toString(), object.toString());
+    }
+
+    vector1.close();
+    vector2.close();
+    buffer1.release();
+  }
+
+  @Test
+  public void testNullableVarCharVectorLoad() {
+    final MaterializedField field = MaterializedField.create(EMPTY_SCHEMA_PATH, NullableVarCharHolder.TYPE);
+
+    // Create a new value vector for 1024 nullable variable length strings.
+    final NullableVarCharVector vector1 = new NullableVarCharVector(field, allocator);
+    final NullableVarCharVector.Mutator mutator = vector1.getMutator();
+    vector1.allocateNew(1024 * 10, 1024);
+
+    // Populate the vector.
+    final StringBuilder stringBuilder = new StringBuilder();
+    final int valueCount = 10;
+    for(int i = 0; i < valueCount; ++i) {
+      stringBuilder.append('x');
+      mutator.set(i, stringBuilder.toString().getBytes(utf8Charset));
+    }
+
+    // Check the contents.
+    final NullableVarCharVector.Accessor accessor1 = vector1.getAccessor();
+    stringBuilder.setLength(0);
+    for(int i = 0; i < valueCount; ++i) {
+      stringBuilder.append('x');
+      final Object object = accessor1.getObject(i);
+      assertEquals(stringBuilder.toString(), object.toString());
+    }
+
+    mutator.setValueCount(valueCount);
+    assertEquals(valueCount, vector1.getAccessor().getValueCount());
+
+    // Still ok after setting value count?
+    stringBuilder.setLength(0);
```

Comment:
do we really need this check on vector1 as we already check this on vector2? we should consider removing this.